clients/unshaded/src/main/java/tachyon/client/block/RemoteBlockInStream.java

```diff
@@ -46,17 +49,37 @@
    * @param blockSize the block size
    * @param location the location
    */
-  // TODO(calvin): Modify the locking so the stream owns the lock instead of the data server.
-  public RemoteBlockInStream(long blockId, long blockSize, NetAddress location) {
+  public RemoteBlockInStream(long blockId, long blockSize, NetAddress location)
+      throws IOException {
     mBlockId = blockId;
     mContext = BlockStoreContext.INSTANCE;
     mBlockSize = blockSize;
+    mClosed = false;
     // TODO(calvin): Validate these fields.
     mLocation = new InetSocketAddress(location.getHost(), location.getDataPort());
+    mWorkerClient = mContext.acquireWorkerClient(location.getHost());
+    String blockPath = mWorkerClient.lockBlock(blockId);
```

Comment:
This should be wrapped in a try block in order to prevent an exception here leaking the `workerClient` resource.